cmd/account/iam-cred-rotate.go

```diff
@@ -0,0 +1,1325 @@
+	if err != nil {
+		return "", fmt.Errorf("error fetching cluster resources: %w", err)
+	}
+	respBody := liveResponse.Body().Resources()
+	if awsAccountClaim, ok := respBody["aws_account_claim"]; ok {
+		var claimJson map[string]interface{}
+		err := json.Unmarshal([]byte(awsAccountClaim), &claimJson)
+		if err != nil {
+			return "", fmt.Errorf("err parsing account claim: %w", err)
+		}
+		if metaData, ok := claimJson["spec"]; ok {
+			if rawClaimName, ok := metaData.(map[string]interface{})["accountLink"]; ok {
+				claimName, ok := rawClaimName.(string)
+				if !ok {
+					return "", fmt.Errorf("parsed account claim metadata name is not a string")
+				}
+				o.claimName = claimName
+				o.log.Info(o.ctx, "Got Cluster AWS Claim:%s\n", o.claimName)
+				return claimName, nil
+			}
+		}
+		return "", fmt.Errorf("unable to get account claim name from JSON")
+	}
+	return "", fmt.Errorf("cluster does not have AccountClaim")
+}
+
+func (o *rotateCredOptions) connectClusterClient() error {
+	o.log.Info(o.ctx, "Connect BP client cluster:'%s'\n", o.clusterID)
+	if len(o.reason) <= 0 {
+		return fmt.Errorf("accessing cred secrets requires a reason for elevated command")
+	}
+	// This action requires elevation
+	elevationReasons := []string{}
+	elevationReasons = append(elevationReasons, o.reason)
+	elevationReasons = append(elevationReasons, fmt.Sprintf("Elevation required to rotate secrets '%s' aws-account-cr-name", o.claimName))
+
+	// If some elevationReasons are provided, then the config will be elevated with user backplane-cluster-admin...
+	// Fetch the backplane kubeconfig, ignore the returned kubecli for now...
+	_, kubeConfig, kubClientSet, err := common.GetKubeConfigAndClient(o.clusterID, elevationReasons...)
+	if err != nil {
+		o.log.Warn(o.ctx, "Err creating cluster:'%s' client, GetKubeConfigAndClient() err:'%+v'\n", o.clusterID, err)
+		return err
+	}
+	o.clusterClientset = kubClientSet
+	// Create new kube client with the CredentialsRequest schema using the backplane config from above...
+	scheme := runtime.NewScheme()
+	err = credreqv1.AddToScheme(scheme)
+	if err != nil {
+		o.log.Warn(o.ctx, "Error adding CredentialsRequests to schema:'%s'\n", err)
+		return err
+	}
+	kubeCli, err := client.New(kubeConfig, client.Options{Scheme: scheme})
+	if err != nil {
+		o.log.Warn(o.ctx, "Error creating new schema'd client for cluster:'%s', err:'%s'\n", o.clusterID, err)
+		return err
+	}
+	o.clusterKubeClient = kubeCli
+	return nil
+}
+
+/* fetch and connect to hive cluster for the provided user cluster */
+func (o *rotateCredOptions) connectHiveClient() error {
+	// This requires a valid ocm login token on 'prod'.
+	// TODO: Can this differ from the login token needed earlier if the cluster
+	// lives in another env (ie integration or staging) (?) If so this needs to allow a 2 step process with
+	// multiple ocm login tokens(?), 1 for each env.
```

Comment:
If hive access (prod) is required, this may need more work to allow testing against a cluster running on staging/integration.  The utils seem to assume a single ocm config context at the moment (?).

Reply:
Has been addressed in recent commit(s) allowing a separate hive ocm config from the host cluster's ocm env/config.